src/test/java/it/eng/idsa/dataapp/web/rest/IncomingDataAppResourceOverWsTest.java

```diff
@@ -0,0 +1,120 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.beans.PropertyChangeEvent;
+import java.util.Map;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import de.fraunhofer.iais.eis.DescriptionResponseMessage;
+import de.fraunhofer.iais.eis.DescriptionResponseMessageBuilder;
+import it.eng.idsa.dataapp.handler.DataAppMessageHandler;
+import it.eng.idsa.dataapp.handler.DescriptionRequestMessageHandler;
+import it.eng.idsa.dataapp.handler.MessageHandlerFactory;
+import it.eng.idsa.dataapp.service.ThreadService;
+import it.eng.idsa.dataapp.util.MessageUtil;
+import it.eng.idsa.multipart.domain.MultipartMessage;
+import it.eng.idsa.multipart.processor.MultipartMessageProcessor;
+import it.eng.idsa.multipart.util.UtilMessageService;
+import it.eng.idsa.streamer.WebSocketServerManager;
+
+class IncomingDataAppResourceOverWsTest {
+
+	@InjectMocks
+	private IncomingDataAppResourceOverWs controller;
+	@Mock
+	private MessageUtil messageUtil;
+	@Mock
+	private MessageHandlerFactory factory;
+	@Mock
+	private DescriptionRequestMessageHandler handler;
+	@Mock
+	private ThreadService threadService;
+	@Mock
+	private Map<String, Object> responseMap;
+	@Mock
+	private PropertyChangeEvent evt;
+	@Mock
+	private WebSocketServerManager webSocketServerManager;
+	MultipartMessage multipartMessage;
+	String requestMessageMultipart;
+
+	@BeforeEach
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+		requestMessageMultipart = createReceivedMessage();
+		multipartMessage = createMultipartMessage();
+		when(threadService.getThreadLocalValue("wss")).thenReturn(true);
+		when(evt.getNewValue()).thenReturn(requestMessageMultipart);
+		when(factory.createMessageHandler(any())).thenReturn(handler);
+		when(handler.handleMessage(multipartMessage.getHeaderContent(), multipartMessage.getPayloadContent()))
+				.thenReturn(responseMap);
+		when(responseMap.get(DataAppMessageHandler.PAYLOAD)).thenReturn("payload");
+	}
+
+	@Test
+	void propertyExchangeTest() {
+		when(responseMap.get(DataAppMessageHandler.HEADER)).thenReturn(getDescriptionResponseMessage());
+		controller.propertyChange(evt);
+		
+		verify(threadService).setThreadLocalValue("wss", true);
+		verify(handler).handleMessage(any(), any());
+	}
+
+	@Test
+	void propertyExchangeExceptionTest() {
+		when(responseMap.get(DataAppMessageHandler.HEADER)).thenReturn("test");
+
+		Exception exception = assertThrows(Exception.class, () -> {
+			controller.propertyChange(evt);
+		});
+	
+		assertNotNull(exception);
+	}
+
+	private MultipartMessage createMultipartMessage() {
+
+		return MultipartMessageProcessor.parseMultipartMessage(requestMessageMultipart);
+	}
+
+	private String createReceivedMessage() {
+
+		return "--JF-UsWbL9rqVxjbb20vdhBSlOT3FiZImD8\r\n" + "Content-Disposition: form-data; name=\"header\"\r\n"
```

Comment:
Can we use one of the Util methods to create MultipartMessage and then convert it to String?

Reply:
It is refactored :)